File: Exercise/Very Hard/08. Poker Hand Ranking/Code/Poker Hand Ranking.py
def get_face(c):
    if len(c) == 2:
        if '2' <= c[0] <= '9':
            return int(c[0])
        elif c[0] == 'J':
            return 11
        elif c[0] == 'Q':
            return 12
        elif c[0] == 'K':
            return 13
        elif c[0] == 'A':
            return 1
    else:
        return 10
    return 0

def get_suit(c):
    if c[-1] == 'h':
        return 3
    elif c[-1] == 'd':
        return 2
    elif c[-1] == 's':
        return 1
    elif c[-1] == 'c':
        return 0
    return 0

def poker_hand_ranking(hand):
    face_count = [0] * 14
    suit_count = [0] * 4

    for card in hand:
        tface = get_face(card)
        tsuit = get_suit(card)

        face_count[tface] += 1
        suit_count[tsuit] += 1

    return ""

if __name__ == "__main__":
    hand = []
    for i in range(5):
        card = input().strip()
        hand.append(card)

    result = poker_hand_ranking(hand)
    print(result)

File: Exercise/Very Hard/08. Poker Hand Ranking/Code/Poker Hand Ranking.java
import java.util.*;

public class Main {
    static int getFace(String c) {
        if (c.length() == 2) {
            if (c.charAt(0) >= '2' && c.charAt(0) <= '9')
                return c.charAt(0) - '0';
            else if (c.charAt(0) == 'J')
                return 11;
            else if (c.charAt(0) == 'Q')
                return 12;
            else if (c.charAt(0) == 'K')
                return 13;
            else if (c.charAt(0) == 'A')
                return 1;
        } else
            return 10;
        return 0;
    }

    static int getSuit(String c) {
        switch (c.charAt(c.length() - 1)) {
            case 'h':
                return 3;
            case 'd':
                return 2;
            case 's':
                return 1;
            case 'c':
                return 0;
        }
        return 0;
    }

    static String pokerHandRanking(List<String> hand) {
        int[] face_count = new int[14];
        int[] suit_count = new int[4];

        Arrays.fill(face_count, 0);
        Arrays.fill(suit_count, 0);

        for (String card : hand) {
            int tface = getFace(card);
            int tsuit = getSuit(card);

            face_count[tface]++;
            suit_count[tsuit]++;
        }

        return "";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<String> hand = new ArrayList<>();

        for (int i = 0; i < 5; ++i) {
            String card = scanner.next();
            hand.add(card);
        }

        String result = pokerHandRanking(hand);
        System.out.println(result);
    }
}




